cfme/tests/openstack/cloud/test_volumes.py

```diff
@@ -0,0 +1,181 @@
+"""Tests for cloud provider's volumes"""
+
+import cfme.web_ui.flash as flash
+import pytest
+from fauxfactory import gen_alphanumeric
+from cfme.cloud.instance import Instance
+from cfme.cloud.volume import (attach_inst_page_form, attach_vlm_page_form,
+                               creation_form, detach_inst_page_form,
+                               detach_vlm_page_form, get_volume_name, list_tbl,
+                               Volume)
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui import InfoBlock, toolbar as tb
+from cfme.web_ui.form_buttons import (add_ng, attach_volume, detach_volume,
+                                      submit_changes)
+from random import choice
+from utils import testgen
+from utils.appliance.implementations.ui import navigate_to
+from utils.version import current_version
+from utils.wait import wait_for
+
+pytest_generate_tests = testgen.generate(testgen.provider_by_type,
+                                         ['openstack'], scope='module')
+
+pytestmark = [pytest.mark.uncollectif(lambda: current_version() >= '5.7'),
+              pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+def test_create_volume(provider):
+    """Creates a volume for given cloud provider"""
```

Comment:
Well, maybe it is better to move 'create_volume' method to global scope as function in volume.py
I think it's more obvious since we can't call 'create_volume' on Volume object because we have no volume yet